includes/Linker.php

```diff
@@ -464,7 +464,11 @@ static function normaliseSpecialPage( Title $title ) {
 	 */
 	private static function fnamePart( $url ) {
 		if (VignetteRequest::isVignetteUrl($url)) {
-			return VignetteRequest::getImageFilename($url);
+			try {
+				return VignetteRequest::getImageFilename($url);
+			} catch (Exception $e) {
+				// VignetteUrlToUrlGenerator logs the exact error
```

Comment:
An exception is thrown so that we know something is wrong, rather than returning null or an empty response. This could have been 

```php
try {
   return VignetteRequest::getImageFilename($url);
} catch (Exception $e) {
  // do the rest of the stuff below
}
```

But that would result in a lot of messages going to ES.

Maybe I should move the try/catch one layer down into the `getImageFilename` function. What do you think of that?